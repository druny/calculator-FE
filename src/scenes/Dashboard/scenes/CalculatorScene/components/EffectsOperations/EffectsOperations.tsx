import React, { Dispatch } from 'react';
import {
  Delete as DeleteIcon,
  DragHandle as DragHandleIcon,
  SvgIconComponent,
} from '@material-ui/icons';

import { CLEAR, EQUAL } from '@/constants';

import { OperationButton } from '../OperationButton';

type EffectType = {
  effectName: string;
  effectLogo: SvgIconComponent;
};

type EffectsOperationsProps = {
  performEffect: Dispatch<string>;
};

const effects: EffectType[] = [
  {
    effectName: EQUAL,
    effectLogo: DragHandleIcon,
  },
  {
    effectName: CLEAR,
    effectLogo: DeleteIcon,
  },
];

export function EffectsOperations({ performEffect }: EffectsOperationsProps) {
  return (
    <div>
      {effects.map(({ effectName, effectLogo }) => (
        <OperationButton
          key={effectName}
          operationLogo={effectLogo}
          onClick={() => performEffect(effectName)}
        />
      ))}
    </div>
  );
}
