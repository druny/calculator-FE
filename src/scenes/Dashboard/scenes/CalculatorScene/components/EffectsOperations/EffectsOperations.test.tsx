import React from 'react';
import renderer from 'react-test-renderer';

import { EffectsOperations } from './EffectsOperations';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<EffectsOperations performEffect={jest.fn()} />).toJSON();
  expect(tree).toMatchSnapshot();
});
