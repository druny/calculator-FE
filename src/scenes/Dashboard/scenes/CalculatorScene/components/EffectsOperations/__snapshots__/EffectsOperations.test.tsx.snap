// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 9H4v2h16V9zM4 15h16v-2H4v2z"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </span>
  </button>
</div>
`;
